src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -170,9 +184,50 @@ public JSONObject getMatchingCourses(HttpServletRequest request) {
       }
     }
     List<Entity> results = db.prepare(courseQuery).asList(FetchOptions.Builder.withDefaults());
+
     return results;
   }
 
+  /* Does substring matching through regex. */
+  private List<Entity> processResults(HttpServletRequest request, List<Entity> fuzzyResults) {
+    List<Entity> processed = new ArrayList<>();
+    Boolean hasName = !request.getParameter("course-name").isEmpty();
+    Boolean hasProf = !request.getParameter("prof-name").isEmpty();
+    Pattern namePattern = Pattern.compile("/(?=a)b/"); // Always false.
```

Comment:
Since these are only used if they get reassigned you can simply initialize here (`Pattern namePattern;`) without assigning.

Reply:
it gives me an error as "namePattern and profPattern may not have been initialized"

Reply:
Ooh right, cause you assign in an if statement... Hm... I suppose this is alright then.